src/app/shared/services/skin/skin.service.ts

```diff
@@ -44,6 +58,11 @@ export class SkinService {
       if (!isInit) {
         this.templateService.initialiseDefaultFieldAndGlobals();
       }
+      // If setTheme OR if the current theme is not available in the target skin,
+      // then set the theme to the targetSkin's default theme
+      if (setTheme || !this.isCurrentThemeAvailableInTargetSkin(targetSkin)) {
```

Comment:
nit(non-blocking)
ideally this should be moved into its own method, e.g. `applySkinThemeChanges` (or similar) which can handle all the conditional and application logic. That way eventually the setSkin method can just call a list of knock-on effectors without having to worry too much about following conditional logic, and individual methods can manage the rest